library/core/src/iter/traits/iterator.rs

```diff
@@ -3382,6 +3382,19 @@ pub trait Iterator {
     ///
     /// assert_eq!(sum, 6);
     /// ```
+    ///
+    /// `sum()` can also be used when iterating over [`Option<T>`][`Option`]
```

Comment:
Maybe it's better to explain that this is a general mechanism (things that implement `Sum`) and that Option and Result are two common examples beyond the number types?